src/main/java/org/apache/sling/jcr/repoinit/impl/AclUtil.java

```diff
@@ -203,6 +204,14 @@ public static void setPrincipalAcl(Session session, String principalName, Collec
         }
 
         final PrincipalAccessControlList acl = getPrincipalAccessControlList(acMgr, principal);
+        if (acl == null) {
```

Comment:
IMO it would be good to explain why acl can be null - best might be to add a comment to the `getPrincipalAccessControlList` to explain that? Or maybe add a `failIfNull` parameter to that method and move the exception building there, so that if we used it from elsewhere we'll be aware of that possible null condition?

The explanation as to why acl can be null was given by Konrad in SLING-10281

Reply:
`getPrincipalAccessControlList` is a private method and only called from here, therefore leaving the exception handling here seems fine. Also I wouldn't do more changes in this context, this was primarily to revert SLING-9449 so the null check has always been here!
I will add a javadoc to that method then.

Reply:
Done in https://github.com/apache/sling-org-apache-sling-jcr-repoinit/pull/14/commits/f0c9a259af9781c8bcc1ca17c84fb62e5571afe8. The rest is IMHO clear from the javadoc of http://jackrabbit.apache.org/api/2.18/index.html?org/apache/jackrabbit/api/security/authorization/PrincipalAccessControlList.html